service.php

```diff
@@ -263,6 +258,25 @@ function get_credentials() {
 		return apply_filters( 'keyring_service_credentials', $creds, $this->get_name() );
 	}
 
+		/**
+	 * Set the basic credential properties based
+	 * on an Array of credentials
+	 *
+	 * @param Mixed $credentials
+	 */
+	function init_credentials( $credentials = null ) {
+		if( ! isset ( $credentials ) ) {
```

Comment:
You could probably just rely an `$credentials` being falsey here.
```suggestion
		if( ! $credentials ) {
```

or use `empty` to cover things like empty arrays:

```suggestion
		if( empty( $credentials ) ) {
```